src/XML/CanonicalizableInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\XMLSecurity\XML;
+
+use SimpleSAML\XML\XMLElementInterface;
+
+/**
+ * An interface for objects that can be canonicalised.
+ *
+ * Objects implementing this interface should retain the original XML structure that was used to create them with
+ * fromXML(), in order to guarantee that the original canonicalisation of the object is the same as the one produced
+ * by this interface.
+ *
+ * If the original DOM object is retained, please remember to implement the \Serializable interface so that the
+ * result of serialisation is the original string representation of the object, since DOM objects are not serializable.
```

Comment:
SerialiZation

Reply:
Excuse me Sir, I just happen to be taught British English 😝